oauth2-authorization-server/src/test/java/org/springframework/security/oauth2/server/authorization/client/TestRegisteredClients.java

```diff
@@ -64,4 +64,20 @@ public static RegisteredClient.Builder registeredPublicClient() {
 				.scope("scope1")
 				.clientSettings(ClientSettings.builder().requireProofKey(true).build());
 	}
+	
+	public static RegisteredClient.Builder registeredClient4() {
```

Comment:
Please remove this and reuse `TestRegisteredClients.registeredClient()`, which returns the `Builder` and allows you to update `clientSecretExpiresAt()`